core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -86,34 +63,27 @@ object Scheduler {
     * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
     */
   def asSourceAndSink(client: MesosClient, podRecordRepository: PodRecordRepository)(implicit mat: Materializer)
-    : (Future[StateSnapshot], Source[StateEventOrSnapshot, NotUsed], Sink[SchedulerCommand, Future[Done]]) = {
-    val flow = fromClient(client, podRecordRepository)
-    asSourceAndSink(flow)(mat)
+    : Future[(StateSnapshot, Source[StateEvent, NotUsed], Sink[SchedulerCommand, Future[Done]])] = {
+    fromClient(client, podRecordRepository).map {
+      case (snapshot, flow) =>
+        val (source, sink) = asSourceAndSink(flow)(mat)
+        (snapshot, source, sink)
+    }(CallerThreadExecutionContext.context)
   }
 
-  def asSourceAndSink(schedulerFlow: Flow[SchedulerCommand, StateOutput, NotUsed])(implicit mat: Materializer)
-    : (Future[StateSnapshot], Source[StateEventOrSnapshot, NotUsed], Sink[SchedulerCommand, Future[Done]]) = {
+  def asSourceAndSink[A, B](schedulerFlow: Flow[A, B, NotUsed])(
```

Comment:
I thinks this should be private or moved out of `Scheduler.scala` since it's not part of the scheduler factory methods that we want to expos.
```suggestion
  def asSourceAndSink[A, B](flow: Flow[A, B, NotUsed])(
```

Since it's generic we probably should adapt the names as well.